Sources/Sentry/include/TracesSampler.h

```diff
@@ -0,0 +1,36 @@
+#import "Random.h"
+#import "SentrySampleDecision.h"
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@class SentryOptions, SentrySamplingContext;
+
+@interface TracesSampler : NSObject
+
+/**
+ *  A random number generator
+ */
+@property (nonatomic, strong) id<Random> random;
```

Comment:
I'm following the Strategy Pattern suggested by @bruno-garcia.
Since TracesSampler is an internal class I don't think this will hurt.

What do you think?

Reply:
It can be an internal dependency if that's possible here, the ctor that takes the protocol only needs to be used by tests

Reply:
I would prefer to hide it with a test ctor. I'm a fan of only exposing what you need.